java-checks/src/main/java/org/sonar/java/checks/StringPrimitiveConstructorCheck.java

```diff
@@ -53,19 +60,83 @@ protected MethodMatchers getMethodInvocationMatchers() {
       MethodMatchers.create().ofTypes("java.lang.Float").constructor().addParametersMatcher("float").build(),
       MethodMatchers.create().ofTypes("java.lang.Double").constructor().addParametersMatcher("double").build(),
       MethodMatchers.create().ofTypes("java.lang.Boolean").constructor().addParametersMatcher("boolean").build(),
-      MethodMatchers.create().ofTypes("java.math.BigInteger").constructor().addParametersMatcher(STRING).build());
+      MethodMatchers.create().ofTypes(BIG_INTEGER).constructor().addParametersMatcher(STRING).build());
   }
 
   @Override
   protected void onConstructorFound(NewClassTree newClassTree) {
+    if (newClassTree.classBody() != null) {
+      return;
+    }
     if(isBigIntegerPotentiallyBiggerThanLong(newClassTree)) {
       return;
     }
-    reportIssue(newClassTree.identifier(), "Remove this \"" + newClassTree.symbolType().name() + "\" constructor");
+    ((InternalJavaIssueBuilder) ((DefaultJavaFileScannerContext) context).newIssue())
+      .forRule(this)
+      .onTree(newClassTree.identifier())
+      .withMessage("Remove this \"" + newClassTree.symbolType().name() + "\" constructor")
+      .withQuickFix(() -> createQuickFix(newClassTree))
+      .report();
+  }
+
+  private static JavaQuickFix createQuickFix(NewClassTree newClassTree) {
+    if (newClassTree.symbolType().is(STRING)) {
+      return createStringQuickFix(newClassTree);
+    } else if (newClassTree.symbolType().is(BIG_INTEGER)) {
+      return createBigIntegerQuickFix(newClassTree);
+    }
+    return createDefaultQuickFix(newClassTree);
+  }
+
+  private static JavaQuickFix createDefaultQuickFix(NewClassTree newClassTree) {
+    String typeName = newClassTree.symbolType().name();
+    return JavaQuickFix.newQuickFix("Replace with " + typeName + ".valueOf")
+        .addTextEdit(JavaTextEdit.replaceTextSpan(
```

Comment:
since both tokens are included, you can directly call replaceBetweenTree:
```java
.addTextEdit(JavaTextEdit.replaceBetweenTree(
  newClassTree.firstToken(), 
  newClassTree.arguments().openParenToken(),
  typeName + ".valueOf("))
.build();
```